test/index.ts

```diff
@@ -43,3 +50,76 @@ describe('Logger', () => {
     assert.deepEqual(ledger, ['aaa'])
   })
 })
+
+describe('withLogger', () => {
+  let ledger: Array<number>
+  let consolelog: any
+
+  beforeEach(() => {
+    ledger = []
+    consolelog = console.log
+    console.log = (message: any) => ledger.push(message)
+  })
+
+  afterEach(() => {
+    console.log = consolelog
+  })
+
+  describe('sync', () => {
+    test('MonadIO1', () => {
```

Comment:
a single test is enough to check the implementation, you can remove the other tests